clang/lib/Sema/SemaOverload.cpp

```diff
@@ -1422,8 +1422,14 @@ static bool IsOverloadOrOverrideImpl(Sema &SemaRef, FunctionDecl *New,
   // the implicit object parameter are of the same type.
 
   auto NormalizeQualifiers = [&](const CXXMethodDecl *M, Qualifiers Q) {
-    if (M->isExplicitObjectMemberFunction())
+    if (M->isExplicitObjectMemberFunction()) {
+      if (M->getNumParams() > 0) {
+        auto ThisType = M->getParamDecl(0)->getType().getCanonicalType();
```

Comment:
```suggestion
        auto ThisType = M->getFunctionObjectParameterReferenceType().getCanonicalType();
```

Do we actually need to call `getCanonicalType()`?